R/testing_utils.R

```diff
@@ -0,0 +1,71 @@
+
+# [Title]   Build A Library For Testing
+# [DESC]    Installs all packages necessary for testing into a another directory,
+#           preferably a temporary directory. This function will throw a fatal error
+#           if that installation fails.
+# [param]   targetLibPath (string) path to the location of the new directory
+# [return]  boolean TRUE
+.BuildTestLib <- function(targetLibPath){
+
+    ### find PKGNET source dir within devtools::test(), R CMD CHECK, and vignette building
+    # NOTE: this can be fragile. Uncomment the lines with "# [DEBUG]" and run test.sh
+    #       from the repo root if something goes wrong
+
+    # [DEBUG] write("=========", file = "~/repos/thing.txt", append = TRUE)
+    # [DEBUG] write(list.files(getwd(), recursive = TRUE), file = "~/repos/thing.txt", append = TRUE)
+    # [DEBUG] write(paste0("working dir: ", getwd()), file = "~/repos/thing.txt", append = TRUE)
+
+    pkgnetSourcePath <- gsub('/pkgnet.Rcheck/tests/testthat$', replacement = '/pkgnet.Rcheck/00_pkg_src/pkgnet', x = getwd())
+    pkgnetSourcePath <- gsub('/pkgnet.Rcheck/tests$', replacement = '/pkgnet.Rcheck/00_pkg_src/pkgnet', x = pkgnetSourcePath)
+    pkgnetSourcePath <- gsub('/pkgnet.Rcheck/vign_test/pkgnet$', replacement = '/pkgnet.Rcheck/00_pkg_src/pkgnet', x = pkgnetSourcePath)
+    pkgnetSourcePath <- gsub('/pkgnet/vignettes$', replacement = '/pkgnet', x = pkgnetSourcePath)
+    pkgnetSourcePath <- gsub('pkgnet/tests/testthat', replacement = 'pkgnet', x = pkgnetSourcePath)
+
+    # [DEBUG] write(paste0("pkgnet path: ", pkgnetSourcePath), file = "~/repos/thing.txt", append = TRUE)
+    # [DEBUG] write("=========", file = "~/repos/thing.txt", append = TRUE)
+
+    ### packages to be built
+    pkgList <- c(
+        baseballstats = file.path(pkgnetSourcePath, "inst", "baseballstats")
+        , sartre = file.path(pkgnetSourcePath, "inst", "sartre")
+        , pkgnet = pkgnetSourcePath
+    )
+
+    ### Install
+
+    # Figure out where R is to avoid those weird warnings about
+    # 'R' should not be used without a path -- see par. 1.6 of the manual.
+    #
+    # NOTE: R CMD CHECK comes with its own bundled "R" binary which doesn't
+    #       work the same way and causes that error. Just trust me on this.
+    #
+    # NOTE: "Sys.which()" would be the correct, portable way to do this but it
+    # doesn't support matching ALL matches, so for now we'll make it work
+    # on unix-alike operating systems and deal with Windows later
+    #
+    R_LOC <- system("which -a R", intern = TRUE)
+    R_LOC <- R_LOC[!grepl("R_check_bin", R_LOC)][1]
```

Comment:
@jayqi @bburns632 this part was the main culprit. Some other `R` executable gets bundled with R CMD CHECK but doesn't totally work and was the source of that dreaded `see par 1.6` warning.

This thing I did here totally works (for both `devtools::test()` case and `R CMD CHECK --as-cran` case) but is not portable to Windows. We can figure that out later.

Reply:
😱